src/App.tsx

```diff
@@ -1,12 +1,119 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
+import { Button } from './components/Button';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goodsFromServer: Good[];
+  hasRequestError: boolean;
+}
+
+enum RequestType {
+  all = 'all',
+  fiveFirst = 'fiveFirst',
+  red = 'red',
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goodsFromServer: [],
+    hasRequestError: false,
+  };
+
+  request = async (type: RequestType) => {
+    let goods = null;
+
+    switch (type) {
+      case 'all':
+        try {
+          goods = await getAll()
+            .then(goodsFromServer => this.setState({ goodsFromServer }));
+
+          return goods;
+        } catch (error) {
+          this.setState({
+            hasRequestError: true,
+          });
+
+          return null;
+        }
+
+      case 'fiveFirst':
+        try {
+          goods = await get5First()
+            .then(goodsFromServer => this.setState({ goodsFromServer }));
+
+          return goods;
+        } catch (error) {
+          this.setState({
+            hasRequestError: true,
+          });
+
+          return null;
+        }
+
+      case 'red':
+        try {
+          goods = await getRedGoods()
+            .then(goodsFromServer => this.setState({ goodsFromServer }));
+
+          return goods;
+        } catch (error) {
+          this.setState({
+            hasRequestError: true,
+          });
+
+          return null;
+        }
+
+      default:
+        return null;
+    }
+  };
```

Comment:
As for me, this a nice use of enum, quite understandable, but a bit huge 👍

Reply:
It really is, you're right. Now that I've seen your approach I guess I'll refactor it, so that it'd take less space)
Thanks for that insight 🥇